internal/app/go-filecoin/paymentchannel/types.go

```diff
@@ -1,18 +1,43 @@
 package paymentchannel
 
 import (
+	"reflect"
+
 	"github.com/filecoin-project/go-address"
 	"github.com/filecoin-project/specs-actors/actors/builtin/paych"
 )
 
+//go:generate cbor-gen-for ChannelInfo VoucherInfo
+
+// The key for the store is the payment channel's "robust" or "unique" address
+
 // ChannelInfo is the primary payment channel record
+// IDAddr: internal ID address for payment channel actor
+// UniqueAddr: aka RobustAddr, used externally to refer to payment channel
+// Duplicated in data due to need to check for existing payment channel since you can't
+// iterate over key/value pairs in statestore at present
 type ChannelInfo struct {
-	Owner    address.Address // Payout (From) address for this channel, has ability to sign and send funds
-	State    *paych.State
-	Vouchers []*VoucherInfo // All vouchers submitted for this channel
+	UniqueAddr, IDAddr, From, To address.Address
+	LastLane                     uint64
+	Vouchers                     []*VoucherInfo // All vouchers submitted for this channel
+}
+
+// ChannelInfoUndef is an empty ChannelInfo
+var ChannelInfoUndef = &ChannelInfo{}
+
+// Equal is a comparator for two ChannelInfos
+func (ci *ChannelInfo) Equal(ci2 *ChannelInfo) bool {
+	if ci.UniqueAddr == ci2.UniqueAddr &&
+		ci.To == ci2.To &&
+		ci.From == ci2.From &&
+		ci.LastLane == ci2.LastLane &&
+		reflect.DeepEqual(ci.Vouchers, ci2.Vouchers) {
```

Comment:
This seems expensive. Is it really necessary? Is it going to be used in high throughput code?

Reply:
Good point; it's only used to check against a blank channel info, so I'll change this func to reflect this and make it faster.